test/success/V-types.prl

```diff
@@ -69,6 +69,16 @@ Thm IdV/Test1(#l:lvl) : [
   lam ty, a. auto
 ].
 
+Thm IdV/Test2(#l:lvl) : [
+  (->
+   [ty : (U #l kan)]
+   [a : ty]
+   (= ty (coe 0~>1 [x] (IdV x #l ty) a)
+         (coe 0~>1 [_] ty a)))
+] by [
+  lam ty, a. refine eq/intro; reduce; auto
```

Comment:
@jonsterling In this case I have to prevent the structural equality rule from running. Should we revisit #288? I am slightly leaning toward to something like `complete (structure + auto) || (reduce + ...)`, or avoiding treating ``coe`` specially, or something fancier...

Reply:
@favonia I was thinking about it a few weeks ago and I think that it is inevitable that we will need to support backtracking, even if in most cases we can code something up without it... I think it would be worth figuring out how to add backtracking to Dependent LCF; then we could have a `append` tactical like in Isabelle (in Coq I think it is called `+`).

Reply:
Honestly I think that if we don't implement backtracking, we really would need to consider limiting the `auto` tactic to only apply invertible rules—which limits its power by quite a bit.

Reply:
It's very easy to understand how to add backtracking when you don't have validations (as in Isabelle), but I think the algorithm is harder when you do have validations. After this week, I can try to cook something up.

Reply:
This issue is now resolved by my PR #419 which implements backtracking.